client/task_runner_test.go

```diff
@@ -73,15 +73,17 @@ func (m *MockTaskStateUpdater) String() string {
 }
 
 type taskRunnerTestCtx struct {
-	upd      *MockTaskStateUpdater
-	tr       *TaskRunner
-	allocDir *allocdir.AllocDir
-	vault    *vaultclient.MockVaultClient
-	consul   *mockConsulServiceClient
+	upd          *MockTaskStateUpdater
+	tr           *TaskRunner
+	allocDir     *allocdir.AllocDir
+	vault        *vaultclient.MockVaultClient
+	consul       *consul.MockAgent
+	consulClient *consul.ServiceClient
```

Comment:
Removing this use of `*mockConsulServiceClient` is what exposed the task runner bug.

There's still a few tests that use that mock as they inspect the order of Consul operations which the mock makes very easy.